### DISCUSSION TITLE
Stratux external gps

### DISCUSSION THREAD

**Discussion by scubasteve4140** [VirusPilot/stratux-pi4#26] [General]
Just put stratux on my Raspberry pi 4 that has a waveshare sim7600gh m.2 hat. Everything works fine when using a ublox usb for my gps receiver, however, I want to be able to use an external gps source. I am getting NMEA messages from my external gps antenna but Stratux isn’t recognizing/packaging it correctly for Foreflight to use it as a gps source. Is there something I can modify in ssh on Stratux for it to recognize external gps as its primary source? My gps information is coming across ttyUSB1.

**Comment by VirusPilot** [comment]
Your NMEA stream needs to show up as /dev/serialin. A quick hack could be a symlink in rc.local or a more sophisticated way would be to create a new udev rule.

**Reply by scubasteve4140** [reply]
I ended up putting this at the end of the interface template before I seen your message. 

{{end}}
# NMEA Input Configuration
nmea:
  - type: udp
    port: 49002
    address: 192.168.1.10


I will give this a shot in the morning and see what it does. If it doesn't work I'll make a change with your recommendations, thanks for the reply!

**Reply by scubasteve4140** [reply]
I created the symlink and it recognizes my external gps source but I can't get the file to stay saved after it reboots. Foreflight is still showing gps connected in devices but no data is getting through.

**Comment by VirusPilot** [comment]
you need to enable persistent logging temporarily followed by a reboot to apply your changes:
![Bildschirmfoto 2024-10-15 um 07 41 16](https://github.com/user-attachments/assets/3c0ac497-5b49-4d9e-81ca-407dd9a354c4)

then you may switch back
